packages/system/src/hooks/useStateProps.ts

```diff
@@ -0,0 +1,45 @@
+import { useRef } from 'react';
+import isEqual from 'react-fast-compare';
+
+export type ComponentState =
+  | 'hover'
+  | 'focus'
+  | 'active'
+  | 'visited'
+  | 'disabled'
+  | 'readOnly'
+  | 'checked'
+  | 'indeterminate'
+  | 'invalid';
+
+export type StateAttrsKey = `data-${Lowercase<ComponentState>}`;
+
+export type StateAttrs = {
+  [key in StateAttrsKey]?: '';
+};
+
+/**
+ * Given a map of states (e.g. `{ hover: true, focus: false }`) returns an
+ * object that can be used to set state props on a component
+ * (e.g. `[data-hover]` and `[data-focus]`).
+ */
+export const useStateProps = (
+  states: { [key in ComponentState]?: boolean } = {}
```

Comment:
coverage has pointed out that there is no test when {} is specified without states. So the default. Then I asked myself, should I just add the test or check in the hook if states are specified and if not show an error message? Or why should you use the hook without specifying states? What do you mean?

Reply:
But then fix coverage with a test for this, right? 😊

Reply:
hm.. still saying line 27 is not tested. Whats the problem?